File: eventhub-backend-integrated/src/main/java/com/eventhub/service/OrganizerRequestService.java
package com.eventhub.service;

import com.eventhub.dto.request.AdminReviewRequest;
import com.eventhub.dto.request.OrganizerRegisterRequest;
import com.eventhub.dto.response.OrganizerRequestResponse;
import com.eventhub.dto.response.PagedResponse;

public interface OrganizerRequestService {
    OrganizerRequestResponse submit(OrganizerRegisterRequest request);
    PagedResponse<OrganizerRequestResponse> getAll(int page, int size, String status);
    OrganizerRequestResponse approve(Long id, AdminReviewRequest request);
    OrganizerRequestResponse reject(Long id, AdminReviewRequest request);
    long getPendingCount();
}

File: campus-ems-integrated/src/services/organizerRequestService.js
import api from '@/lib/axios';
import EP from '@/lib/endpoints';

export const organizerRequestService = {
  // Public — applicant submits
  submit:       (d)                => api.post(EP.ORGANIZER_REQUESTS.SUBMIT, d),
  // Admin only
  getAll:       (page, size, status) => api.get(EP.ORGANIZER_REQUESTS.LIST, { params: { page, size, status } }),
  getPendingCount: ()              => api.get(EP.ORGANIZER_REQUESTS.PENDING_COUNT),
  approve:      (id, note)         => api.put(EP.ORGANIZER_REQUESTS.APPROVE(id), { adminNote: note }),
  reject:       (id, note)         => api.put(EP.ORGANIZER_REQUESTS.REJECT(id),  { adminNote: note }),
};